plugins/twitter/classes/DataProvider/Twitter.php

```diff
@@ -82,24 +86,56 @@ public function fetch($limit = FALSE) {
 				"result_type" => 'recent'
 			]);
 
-			if ( empty($results->statuses))
+			if ( empty($results['statuses']))
 			{
 				return 0;
 			}
 
-			$statuses = $results->statuses;
+			$statuses = $results['statuses'];
 
 			// Store the highest id
-			$this->since_id = $statuses[0]->id;
+			$this->since_id = $statuses[0]['id'];
 
 			foreach ($statuses as $status) {
-				$id = $status->id;
-				$user = $status->user;
-				$screen_name = $user->screen_name;
-				$text = $status->text;
+				$id = $status['id'];
+				$user = $status['user'];
+				$screen_name = $user['screen_name'];
+				$text = $status['text'];
+
+				$additional_data = [];
+
+				if ($status['coordinates'] || $status['place']) {
+					$additional_data['location'] = [];
+					if ($status['coordinates']) {
+						$additional_data['location'][] = $status['coordinates'];
+					}
+
+					if ($status['place'] && $status['place']['bounding_box']) {
+						// Make a valid linear ring
+						$status['place']['bounding_box']['coordinates'][0][] = $status['place']['bounding_box']['coordinates'][0][0];
+
+						// If we don't already have a location
+						if (empty($additional_data['location'])) {
+							// Find center of bounding box
+							$geom = GeoJSON::geomFromText(json_encode($status['place']['bounding_box']));
+							// Use mysql to run Centroid
+							$result = DB::select([
```

Comment:
This is pretty ugly - but brick/geo was a non trivial change and required PHP 5.6